arangod/Agency/AgencyComm.cpp

```diff
@@ -177,13 +177,56 @@ AgencyOperation::AgencyOperation(std::string const& key,
   _opType.value = opType;
 }
 
+AgencyOperation::AgencyOperation(std::string const& key, AgencyValueOperationType opType,
+                                 std::shared_ptr<velocypack::Builder> value)
+    : AgencyOperation(key, opType, value->slice()) {
+  _holder = std::move(value);
+  TRI_ASSERT(_holder->slice().start() == _value.start());
+}
+
 AgencyOperation::AgencyOperation(std::string const& key, AgencyValueOperationType opType,
                                  VPackSlice newValue, VPackSlice oldValue)
     : _key(AgencyCommHelper::path(key)), _opType(), _value(newValue), _value2(oldValue) {
   _opType.type = AgencyOperationType::Type::VALUE;
   _opType.value = opType;
 }
 
+AgencyOperation::AgencyOperation(std::shared_ptr<cluster::paths::Path const> const& path)
+    : _key(path->str()), _opType() {
+  _opType.type = AgencyOperationType::Type::READ;
+}
+AgencyOperation::AgencyOperation(std::shared_ptr<cluster::paths::Path const> const& path,
+                                 AgencySimpleOperationType opType)
+    : _key(path->str()), _opType() {
+  _opType.type = AgencyOperationType::Type::SIMPLE;
+  _opType.simple = opType;
+}
+
+AgencyOperation::AgencyOperation(std::shared_ptr<cluster::paths::Path const> const& path,
+                                 AgencyValueOperationType opType,
+                                 velocypack::Slice const value)
+    : _key(path->str()), _opType(), _value(value) {
```

Comment:
This constructor is potentially unsafe to use. It only works if the lifetime of the `Slice` in value exceeds the lifetime of the AgencyOperation.
Should we make this ctor protected, or at least add a code comment that indicates that?

Reply:
I agree. Note that these constructors (initializing from a Slice, but not copying it) are the oldest and possibly most used ones, so we can't just protect them without rewriting some code. I've added some comments now. I'll see if I can spare some time to clear all this up, but rather not in this PR if that's fine with you.